py/desitarget/uratmatch.py

```diff
@@ -0,0 +1,612 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+# -*- coding: utf-8 -*-
+"""
+====================
+desitarget.uratmatch
+====================
+
+Useful `URAT`_ matching and manipulation routines.
+
+.. _`URAT`: http://cdsarc.u-strasbg.fr/viz-bin/cat/I/329
+"""
+import os
+import numpy as np
+import fitsio
+import requests
+import pickle
+
+from pkg_resources import resource_filename
+from time import time
+from astropy.io import ascii
+from glob import glob
+import healpy as hp
+
+from desitarget.internal import sharedmem
+from desimodel.footprint import radec2pix
+from desitarget.geomask import add_hp_neighbors, radec_match_to
+
+# ADM set up the DESI default logger
+from desiutil.log import get_logger
+log = get_logger()
+
+# ADM start the clock
+start = time()
+
+# ADM columns contained in our version of the URAT fits files.
+uratdatamodel = np.array([], dtype=[
+    ('URAT_ID', '>i8'), ('RA', '>f8'), ('DEC', '>f8'),
+    ('APASS_G_MAG', '>f4'), ('APASS_G_MAG_ERROR', '>f4'),
+    ('APASS_R_MAG', '>f4'), ('APASS_R_MAG_ERROR', '>f4'),
+    ('APASS_I_MAG', '>f4'), ('APASS_I_MAG_ERROR', '>f4'),
+    ('PMRA', '>f4'), ('PMDEC', '>f4'), ('PM_ERROR', '>f4')
+])
+
+
+def _get_urat_dir():
+    """Convenience function to grab the URAT environment variable.
+
+    Returns
+    -------
+    :class:`str`
+        The directory stored in the $URAT_DIR environment variable.
+    """
+    # ADM check that the $URAT_DIR environment variable is set.
+    uratdir = os.environ.get('URAT_DIR')
+    if uratdir is None:
+        msg = "Set $URAT_DIR environment variable!"
```

Comment:
Let's give the user a hint of where that directory should be when running at NERSC.  Please add it to `etc/desitarget.module` so that our code installations at NERSC will automatically set it.

Reply:
Done.